packages/xo-web/src/common/action-button.js

```diff
@@ -92,9 +94,18 @@ export default class ActionButton extends Component {
 
       const { redirectOnSuccess } = props
       if (redirectOnSuccess !== undefined) {
-        const to = typeof redirectOnSuccess === 'function' ? redirectOnSuccess(result, handlerParam) : redirectOnSuccess
-        if (to !== undefined) {
-          return this.context.router.push(to)
+        switch (typeof redirectOnSuccess) {
+          case 'number':
+            if (redirectOnSuccess === ActionButton.GO_BACK) {
+              return this.context.router.goBack()
+            }
```

Comment:
Do an explicit check first and remove `number` case, it will simplify the code and remove some unnecessary checks.